design-proposals/migration-target.md

```diff
@@ -0,0 +1,97 @@
+The motivation is to better define a boundary between what is an absolute and long-lasting property of a VM (like affinity) with what is just an optional property of the single migration attempt.
+This could also be relevant in terms of personas: we could have the VM owner/developer that is going to specify long-lasting affinity for a VM that is part of an application composed by different VMs and pods and a cluster admin/operator that needs to temporary override that for maintenance reasons.
+On the other side the VM owner is not required/supposed to be aware of node names. 
+
+## Goals
+- A user allowed to trigger a live-migration of a VM and list the nodes in the cluster is able to rely on a simple and direct API to try to live migrate a VM to a specific node.
+- The explict migration target overrules a nodeSelector or affinity and anti-affinity rules defined by the VM owner.
+- The livemigration then can successfully complete or fail for various reasons exactly as it can succeed of fail today for other reasons. 
+- The target node that is explicitly required for the actual livemigration attempt should not influence future livemigrations or the placement in case the VM is restarted. For long-lasting placement, nodeSelectors or affinity/anti-affinity rules are the only way to go. 
+
+## Non Goals
+- this proposal is not defining a custom scheduler plugin nor suggesting to alter how the default k8s scheduler works with `nodeName`, `nodeSelector` and `affinity/anti-affinity` rules. See https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/ for the relevant documentation
+
+## Definition of Users
+- VM owner: the user who owns a VM in his namespace on a Kubernetes cluster with KubeVirt
+- Cluster-admin: the administrator of the cluster
+
+## User Stories
+- As a cluster admin I want to be able to try to live-migrate a VM to specific node for maintenance reasons eventually overriding what the VM owner set 
+- As a VM owner I don't want to see my VM object getting amended by another user just for maintenance reasons
+
+## Repos
+- https://github.com/kubevirt/kubevirt
+
+# Design
+We are going to add a new optional `nodeName` string field on the `VirtualMachineInstanceMigration` object.
+If the `nodeName` filed is not empty, the migration controller will explicitly set `nodeName` on the virt-launcher pod that is going to be used as the target endpoint for the livemigration.
+If the `nodeName` field is not empty, the k8s scheduler will ignore the Pod that is going to be used as the target for the migration and the kubelet on the named node will directly try to place the Pod on that node.
+Using `nodeName` on the `VirtualMachineInstanceMigration` object will overrule nodeSelector or affinity and anti-affinity rules defined on the VM.
+If the target pod will fail to be started, the `VirtualMachineInstanceMigration` object will be marked as failed as it can already happen today for other reasons.
```

Comment:
```suggestion
If the target pod fails to be started, the `VirtualMachineInstanceMigration` object will be marked as failed as it can already happen today for other reasons.
```
English is weird ;-)

Reply:
done, thanks